sdk/eventhub/azure-eventhub/azure/eventhub/aio/_client_base_async.py

```diff
@@ -83,6 +83,24 @@ async def get_token(self, *scopes: str, **kwargs: Any) -> AccessToken:  # pylint
         """
         return AccessToken(self.token, self.expiry)
 
+class AzureNamedKeyTokenCredentialAsync(object):
+    """The named key credential used for authentication.
+
+    :param str credential: The AzureNamedKeyCredential that should be used
+    """
+
+    def __init__(self, credential):
+        # type: (AzureNamedKeyCredential) -> None
+        self.credential = credential
+        self.token_type = b"servicebus.windows.net:sastoken"
+
+    async def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
+        if not scopes:
+            raise ValueError("No token scope provided.")
+        name, key = self.credential.named_key
+        return _generate_sas_token(scopes[0], name, key)
```

Comment:
super nit: coding consistency with `AzureSasTokenCredential`
```suggestion
    def __init__(self, azure_named_key_credential):
        # type: (AzureNamedKeyCredential) -> None
        self._credential = azure_named_key_credential
        self.token_type = b"servicebus.windows.net:sastoken"

    async def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
        if not scopes:
            raise ValueError("No token scope provided.")
        name, key = self._credential.named_key
        return _generate_sas_token(scopes[0], name, key)
```